### ISSUE TITLE
for loops don't allow latedef

### ISSUE THREAD

**Issue by lukeapage** [jshint/jshint#2429]
The following warns that `i` is a global (line 1) and that `i` is unused from line 3.
```
for(i in {}) {
}
var i;
```

**Comment by xak2000** [comment]
I recently updated JSHint (don't know which version was before) and it starts complain to this code:
```
var stompClient = {
      getClient: function () {return client;}
    },
    client;
```
`'client' was used before it was defined.`

This error is shown when `"latedef": "nofunc"` or `"latedef": true` and not shown when `"latedef": false`.

Before there was no error with the same options.
Don't know if it is the same case as `for` loop. If no I can open another issue.

**Comment by nicolo-ribaudo** [comment]
http://jshint.com/docs/options/#latedef
> This option prohibits the use of a variable before it was defined. JavaScript has function scope only and, in addition to that, all variables are always moved—or hoisted— to the top of the function. This behavior can lead to some very nasty bugs and that's why it is safer to always use variable only after they have been explicitly defined.

> Setting this option to "nofunc" will allow function declarations to be ignored.

`client` is defined at line 4 but you are using it before, at line 2.

**Comment by xak2000** [comment]
No, it is not used at line 2. It used in anonymous function defined at line 2. This anonymous function isn't called at line 2. Only defined.

If this were `getClient: client` or `getClient: (function () {return client;})()` then yes, `client` is used before defined.
